pgzero/storage.py

```diff
@@ -0,0 +1,114 @@
+import json
+import os
+import platform
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+__all__ = ['StorageCorruptionException', 'JSONEncodingException', 'Storage']
+
+
+class StorageCorruptionException(Exception):
+    """The data in the storage is corrupted."""
+
+
+class JSONEncodingException(Exception):
+    """The data in the storage is corrupted."""
+
+
+class Storage(dict):
+
+    path = ''
+
+    """Behaves like a dictionary with a few extra functions.
+
+    It's possible to load/save the data to disk.
+
+    The name of the file will be the script's name hashed.
+
+    NOTE: If two scripts have the same name, they will load/save from/to
```

Comment:
I would expect this only if they have exactly the same path in the filesystem, no?

Reply:
I was getting the name from `mod.__file__` so basically, if people use the relative path for the script when calling `pgzrun intro.py` then I would only get `intro.py`.

I corrected this to check if the path is absolute and if it isn't I convert it to absolute, like so:

```
    filename = mod.__file__

    if not os.path.isabs(filename):
        filename = os.path.join(os.getcwd(), filename)

    fn_hash = hashlib.sha1(filename.encode('utf-8'))

    Storage.set_app_hash(fn_hash.hexdigest())
```